dhis-2/dhis-api/src/main/java/org/hisp/dhis/category/CategoryOptionCombo.java

```diff
@@ -162,7 +164,7 @@ public void removeAllCategoryOptions() {
 
   @Override
   public boolean isDefault() {
-    return categoryCombo != null && DEFAULT_NAME.equals(categoryCombo.getName());
+    return DEFAULT_CATEGORY_COMBO_NAME.equals(name);
```

Comment:
I suggest `DEFAULT_NAME.equals(name)` because you're testing a CatOptionCombo name, not a CatCombo name. (They're both named "default" but it's better form to test against the one that's defined for this object.)

Note that the line you're replacing was also not great for the opposite reason; it was comparing a CatCombo name with the default CatOptionCombo name.

Reply:
I used it as that is what is used to find it in https://github.com/dhis2/dhis2-core/blob/2313c47c91d9229ad0659b0ed5818d9ceaad40ba/dhis-2/dhis-services/dhis-service-core/src/main/java/org/hisp/dhis/category/DefaultCategoryService.java#L583

I am making your suggested change 👌🏻 but then also changing `DefaultCategoryService.getDefaultCategoryOptionCombo`.

Reply:
Very good, it would be right to change that too. Functionally it doesn't matter because all the names are "default", but if you do a code search for all uses of `CategoryOptionCombo.DEFAULT_NAME`, you should find all the tests for the CatOptionCombo default name. Likewise, if you search for all uses of `CategoryCombo.DEFAULT_CATEGORY_COMBO_NAME` you should fine all the tests for the CatCombo default name, etc.